pkg/allow/allow_labels.go

```diff
@@ -49,11 +51,23 @@ func (a Labels) Allowed(metric string, labels, values []string) ([]string, []str
 
 	var finalLabels, finalValues []string
 	labelSet := labelSet(allowedLabels)
+	alreadyMatched := make(map[string]interface{}) // used to prevent duplicate labels
 	for allowedLabel := range labelSet {
 		for i, label := range labels {
-			if label == allowedLabel {
+			_, ok := alreadyMatched[label]
+			if ok {
+				continue
+			}
+			allowedLabelReg, err := regexp.Compile(allowedLabel)
+			if err != nil {
+				klog.Error(err)
+				continue
+			}
+
+			if allowedLabelReg.MatchString(label) {
```

Comment:
do we need to regulate this more?  for example, if the user specifies "label1" in the allow list, this would let "label1*" labels through (i.e. "label11", "label111", etc).

Reply:
Since it's just regex, one can specify `^label1$`. WDYT? I don't think we should support implicit `^<LABEL>$`

Reply:
works for me.  can we update the doc(s) to reflect this?

Reply:
I do think this might need some more thought. The example `label1` actually also allows `some_label1`. Implicit `^<LABEL>$` would prevent that from happening but one can still specify `.*label1.*` for the same functionality as it does right now.

Having users require putting in `^label1$` for each exact label essentially creates the same issue as this one attempts to solve: adding more extra information that would have been necessary. Granted, it's probably not too much as most would probably want to use just a few well designed regex.